app/src/apiService.ts

```diff
@@ -24,7 +25,36 @@ export class ApiService implements API {
         this.axiosInstance = axiosInstance;
     }
 
+    private handleDownloadResponse<T>(response: AxiosResponse): T {
+        if (response.headers["content-type"] !== "text/html") {
+            // should render blank on iframe
+            return "" as T;
+        }
+        return URL.createObjectURL(response.data) as T;
+    }
+
+    private handleDownloadError = (error: AxiosError) => {
+        console.log(error.response && error.response.data || error);
+        let errorMessage = {error: {detail: "Could not parse API response", error: "error"}};
+        if (error instanceof AxiosError && error.response) {
+            errorMessage = error.response.data as Error;
+        }
+        return errorMessage;
+    };
```

Comment:
Looks like this is identical to how non-download errors are handled, so you could call this method for both,

Reply:
Absolutely